connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

```diff
@@ -545,8 +544,25 @@ private void expectConvertKeyValue(boolean anyTimes) {
             convertValueExpect.andReturn(SERIALIZED_RECORD);
     }
 
-    private boolean awaitPolls(CountDownLatch latch) throws InterruptedException {
-        return latch.await(1000, TimeUnit.MILLISECONDS);
+    private void expectTaskCommitRecord(boolean anyTimes) throws InterruptedException {
+        sourceTask.commitRecord(EasyMock.anyObject(SourceRecord.class));
+        IExpectationSetters<Void> expect = EasyMock.expectLastCall();
+        boolean succeed = random.nextBoolean();
```

Comment:
This seems like it'll make any test failures transient since you can't rely on consistent behavior from run to run. Why not test the specific cases we want to to evaluate -- either it fails or it doesn't? i.e. why not have a separate test for `commitRecord` failing instead of making it random?